sdks/go/pkg/beam/register/iter.go

```diff
@@ -113,7 +123,9 @@ func Iter1[T any]() {
 	registerFunc := func(s exec.ReStream) exec.ReusableInput {
 		return &iter1[T]{s: s}
 	}
-	exec.RegisterInput(reflect.TypeOf(i).Elem(), registerFunc)
+	itT := reflect.TypeOf(i).Elem()
```

Comment:
Hey @lostluck 
I just started a test run with your changes applied with the bigtable io connector (#23411).
Unfortunately I still ran into the following error: `reflect.Set: value of type struct { RowKey string; Ops []struct { Family string; Column string; Ts int64; Value []uint8 }; GroupKey string } is not assignable to type bigtableio.Mutation`.
I played around with this implementation but wasn't able to make it work. My knowledge on the reflect package is somehow limited though.